src/core/lombok/core/configuration/ConfigurationParser.java

```diff
@@ -36,20 +38,46 @@ public ConfigurationParser(ConfigurationProblemReporter reporter) {
 		this.reporter = reporter;
 	}
 	
-	public void parse(CharSequence content, String contentDescription, Collector collector) {
+	public void parse(ConfigurationFile context, Collector collector) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 22. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.